Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Login & Signup Flow/NonprofitSignupViewController.m

```diff
@@ -68,61 +68,27 @@ - (IBAction)getStartedButtonTapped:(id)sender {
 -(void)saveNonprofitAndUserToParse {
     [self saveNonprofitDataToNonprofitObject];
     
-    NSOperationQueue *opQueue = [NSOperationQueue new];
-
-    NSBlockOperation *signupUserOp = [NSBlockOperation new];
-    signupUserOp.queuePriority = NSOperationQueuePriorityHigh;
-    __weak NSBlockOperation *weakSignupUserOp = signupUserOp;
-    [signupUserOp addExecutionBlock:^{
-        [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-            if (!succeeded) {
-                [opQueue cancelAllOperations];
-                UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not save user." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
-                [self presentViewController:alert animated:YES completion:nil];
-            }
-        }];
-    }];
-    
-    NSBlockOperation *saveNonprofitOp = [NSBlockOperation new];
-    saveNonprofitOp.queuePriority = NSOperationQueuePriorityHigh;
-    __weak NSBlockOperation *weakSaveNonprofitOp = saveNonprofitOp;
-    [saveNonprofitOp addExecutionBlock:^{
-        [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-            if ([weakSaveNonprofitOp isCancelled]) {
-                [self.user deleteInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
-                    if (!succeeded) {
-                        NSLog(@"error");
-                    }
-                }];
-                return;
-            } else if (!succeeded) {
-                [opQueue cancelAllOperations];
-                UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Could not save nonprofit." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
-                [self presentViewController:alert animated:YES completion:nil];
-            }
-        }];
-    }];
-    
-    
-    NSBlockOperation *loginSegueOp = [NSBlockOperation new];
-    //FIXME: loginSegueOp is still running first??
-    loginSegueOp.queuePriority = NSOperationQueuePriorityLow;
-    __weak NSBlockOperation *weakLoginSegueOp = loginSegueOp;
-    [loginSegueOp addExecutionBlock:^{
-        if (weakLoginSegueOp.isCancelled) {
-            return;
+    [self.user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+        if (succeeded) {
+            [self.nonprofit saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
```

Comment:
I think you will have to use `weakSelf` from this point to prevent a retain cycle and memory leak. You can override the `dealloc` method on this view controller and add a log to check if it's called after navigating back to the previous screen, if it's not called then you have a memory leak and need to use `weakSelf` inside all these blocks.